### PR TITLE
Decode more StgTSO fields

### PR DESCRIPTION
**Pull request by supersven**
Stay in sync with the recent additions to `wip/ghc-debug`.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -1,4 +1,4 @@
-packages: common, stub, client, dwarfadt, dwarf, ghc-vis, test, dyepack-test
+packages: common, stub, client, dwarfadt, dwarf, ghc-vis, test, dyepack-test, test, /home/sven/src/head.hackage/packages/io-streams-1.5.1.0
```
Comment (DavidEichmann) [inline cabal.project:1]: Getting things to build is one of my least favourite activities :cry: But I managed to figure out how to get this to build. I think you can get rid of this change and instead copy the head.hackage [constraints](https://ghc.gitlab.haskell.org/head.hackage/cabal.constraints) (at least the one for `io-streams`) into this `cabal.project` file (this is described on the [head.hackage page](https://ghc.gitlab.haskell.org/head.hackage/))

Comment (supersven) [reply cabal.project:1]: Ups, Sorry!

`/home/sven/src/head.hackage/packages/io-streams-1.5.1.0` is a left over from the time when I was waiting for my `io-streams` patch being merged to `head.hackage`. (https://gitlab.haskell.org/ghc/head.hackage/-/merge_requests/111)

Comment (DavidEichmann) [reply cabal.project:1]: Ah great, so the head.hackage patch is already merged. Still though, when I did `cabal new-build` it selected the wrong is-streams version. Perhaps adding the constraints is not a bad idea (though maybe a separate PR).

**Comment by supersven** [comment]
Hey @DavidEichmann ,
I'll solve the review comments/todos in the GHC branch first (https://gitlab.haskell.org/ghc/ghc/-/merge_requests/1435) - otherwise I would get back into some "ping-pong-mode" that didn't prove to be very efficient...

As soon as we solved all open issues on the ghc-debug branch (or at least all API changes), I'll update this branch.

File: cabal.project (hunk 1 above)
Comment (DavidEichmann) [reply cabal.project:1]: See here for more build instructions https://github.com/bgamari/ghc-debug/issues/42#issuecomment-693359269

**Pull request merged**